src/processor/mapper/map_extend.cpp

```diff
@@ -160,19 +159,27 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapLogicalRecursiveExtendToPhysica
             tmpSrcNodePos, std::vector<DataPos>{tmpDstNodePos}, std::move(scanFrontier),
             getOperatorID(), emptyParamString);
     }
+    std::unique_ptr<RecursiveJoinDataInfo> dataInfo;
+    if (extend->trackPath()) {
+        auto pathVectorPos = DataPos(outSchema->getExpressionPos(*rel));
+        dataInfo = std::make_unique<RecursiveJoinDataInfo>(outDataPoses, colIndicesToScan,
+            inNodeIDVectorPos, outNodeIDVectorPos, tmpDstNodePos, true /* trackPath */,
```

Comment:
I choose to not change this to enum yet because for a very long time we will just need to check between track/not track. So we probably don't need an ENUM with 2 cases.